packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/Controls.tsx

```diff
@@ -73,92 +66,20 @@ export const LocationDetailViewControls = ({
   LocationDetailViewProps,
   'children' | 'className'
 >): React.JSX.Element => {
-  const [{ data, isLoading, hasError }, handleList] = useAction(
-    'LIST_LOCATION_ITEMS'
-  );
-
-  const [{ history }, dispatchStoreAction] = useStore();
-  const { current } = history;
-  const { prefix } = current ?? {};
-  const hasInvalidPrefix = isUndefined(prefix);
-
-  const handleDroppedFiles = (files: File[]) => {
-    dispatchStoreAction({ type: 'ADD_FILE_ITEMS', files });
-    dispatchStoreAction({
-      type: 'SET_ACTION_TYPE',
-      actionType: 'UPLOAD_FILES',
-    });
-
-    if (isFunction(onActionSelect)) onActionSelect('UPLOAD_FILES');
-  };
-
-  const { result, nextToken } = data;
-  const resultCount = result.length;
-  const hasNextToken = !!nextToken;
-
-  const onPaginateNext = () => {
-    if (hasInvalidPrefix || !nextToken) return;
-    dispatchStoreAction({ type: 'RESET_LOCATION_ITEMS' });
-    handleList({ prefix, options: { ...DEFAULT_LIST_OPTIONS, nextToken } });
-  };
-
-  const onPaginatePrevious = () => {
-    dispatchStoreAction({ type: 'RESET_LOCATION_ITEMS' });
-  };
-
+  const locationDetailView = useLocationDetailView({ onNavigate });
   const {
-    currentPage,
-    handlePaginateNext,
-    handlePaginatePrevious,
-    handleReset: handlePaginateReset,
-  } = usePaginate({
-    pageSize: DEFAULT_PAGE_SIZE,
-    onPaginateNext,
-    onPaginatePrevious,
-  });
-
-  React.useEffect(() => {
-    if (hasInvalidPrefix) return;
-
-    handleList({ prefix, options: DEFAULT_REFRESH_OPTIONS });
-
-    handlePaginateReset();
-  }, [
-    dispatchStoreAction,
-    handleList,
-    handlePaginateReset,
-    hasInvalidPrefix,
-    prefix,
-  ]);
-
-  const {
-    disableActionsMenu,
-    disableNext,
-    disablePrevious,
-    disableRefresh,
-    range,
-    renderLoading,
-  } = listViewHelpers({
-    currentPage,
-    hasNextToken,
+    pageItems,
     isLoading,
-    pageSize: DEFAULT_PAGE_SIZE,
-    resultCount,
-    hasError,
-  });
-
-  const handleRefresh = () => {
-    if (hasInvalidPrefix) return;
-    handlePaginateReset();
-    handleList({ prefix, options: DEFAULT_REFRESH_OPTIONS });
-    dispatchStoreAction({ type: 'RESET_LOCATION_ITEMS' });
-  };
+    page,
+    isPaginatePreviousDisabled,
+    isPaginateNextDisabled,
+  } = locationDetailView;
 
   const contextValue: ControlsContext = {
     data: {
-      isDataRefreshDisabled: disableRefresh,
+      isDataRefreshDisabled: isLoading,
     },
-    onRefresh: handleRefresh,
+    onRefresh: locationDetailView.onRefresh,
```

Comment:
This can just be destructured from the hook right?
```suggestion
    onRefresh,
```